js/web/plunderer/js/plunderer.js

```diff
@@ -508,17 +562,18 @@ let Plunderer = {
 	},
 
 
-	RenderActions: (actions) => {
+	RenderActions: async (actions) => {
 		let lastPlayerId = null;
-		return actions.map(action => {
+		let actionRenders = await Promise.all(actions.map(async (action) => {
 			const isSamePlayer = action.playerId === lastPlayerId;
 			lastPlayerId = action.playerId;
 			return Plunderer.RenderAction({action, isSamePlayer});
-		}).join('');
+		}));
+		return actionRenders.join('');
 	},
 
 
-	RenderAction: ({action, isSamePlayer}) => {
+	RenderAction: async ({action, isSamePlayer}) => {
```

Comment:
same fere, I prefer to enrich action with "factor",  like player's info:

```js
		// enrich actions with player info
		const players = await Plunderer.db.players.where('id').anyOf(actions.map(it => it.playerId)).toArray();
		actions = actions.map(it => {
			const player = players.find(p => p.id === it.playerId);
			const playerFromDict = PlayerDict[it.playerId];
			// Try get info about player from indexdb, if not possible than from PlayerDict
			const playerInfo = player ? ({
				playerName: player.name,
				avatar: player.avatar,
				playerDate: player.date,
				clanId: player.clanId,
				clanName: player.clanName || i18n('Boxes.Plunderer.HasNoClan'),
				playerEra: player.era
			}) : playerFromDict ? ({
				playerName: playerFromDict.PlayerName,
				clanId: playerFromDict.ClanId || 0,
				clanName: playerFromDict.ClanName || i18n('Boxes.Plunderer.HasNoClan'),
				avatar: playerFromDict.Avatar,
				playerEra: 'unknown',
				playerDate: null,
			}) : ({
				playerName: 'Unknown',
				clanId: 'N/A',
				clanName: 'Unknown',
				avatar: null,
				playerEra: 'unknown',
				playerDate: null,
			});
			return {
				...it,
				...playerInfo,
			};
		});
```

Below code does not use Key-value lookup table, it should be iptimized one day, and need to replace `const player = players.find(p => p.id === it.playerId);` within lookup table